### DISCUSSION TITLE
Indeterminate state of checkbox not visible in tree view

### DISCUSSION THREAD

**Discussion by taruntadikonda** [primefaces/community#317] [PrimeNG]
Indeterminate state of checkbox not visible in tree when selectionMode as checkbox and a leaf node is assigned in the selection property(ngModel).

![image](https://github.com/primefaces/community/assets/25893079/66f37ef7-4a7c-40fd-9d40-bd7b306152a0)

In the above image 1.1.2 is assigned to selection property of tree but the indeterminate state of checkbox of the parent nodes are not visible.
If I explicitly check the label 1.1.2, I can see the indeterminate state of the checkbox in the parent nodes.

![image](https://github.com/primefaces/community/assets/25893079/f5ff6f55-4b37-4c8a-b3dc-85c872d436ee)

html file
`  <p-tree [value]="files" selectionMode="checkbox" class="w-full md:w-30rem" [(selection)]="selectedFiles"></p-tree>
`

ts file

`import { Component } from '@angular/core';`
`import {data} from "./data"`

`@Component({`
  `selector: 'app-root',`
  `templateUrl: './app.component.html',`
  `styleUrls: ['./app.component.scss']`
`})`
`export class AppComponent {`
  `files=data`
  `selectedFiles: any=[ {`
  `  key: '0-0-1',`
  `  label: '1.1.2'`
 ` }];`
`}`

data file

`export const data = [`
`  {`
    `key: '0',`
   ` label: '1',`
    `expanded:true,`
    `children: [`
     ` {`
       ` key: '0-0',`
       ` label: '1.1',`
       ` expanded:true,`
       ` children: [`
         ` {`
          `  key: '0-0-0',`
          `  label: '1.1.1',`
       `   },`
        `  {`
           ` key: '0-0-1',`
           ` label: '1.1.2',`
        `  },`
   `     ],`
    `  },`
  `  ],`
 ` },`
`];`

**Comment by cetincakiroglu** [comment]
Hi,

Could you please create a Stackblitz reproducer so we can identify the issue clearly?

**Reply by taruntadikonda** [reply]
https://stackblitz.com/edit/stackblitz-starters-wubfjm?file=src%2Fglobal_styles.css

**Comment by cetincakiroglu** [comment]
Hi,

Thanks for reporting the issue, it seems that there is a bug with the intermediate checkbox, we've opened the issue in the repo you can track https://github.com/primefaces/primeng/issues/13985
